### PR TITLE
add cross compile windows linux and darwin to be built

### PR DESCRIPTION
**Pull request by bmanojlovic**
Except already stated XC targets, added stripping of produced binaries

### CODE REVIEW

File: example/golang.mk [hunk 1]
```diff
@@ -14,6 +14,7 @@ BUILD_MACHINE=$(shell echo $$HOSTNAME)
 BUILD_USER=$(shell whoami)
 
 BUILD_FLAGS=-ldflags "\
+	-s -w \
```
Comment (svenwltr) [inline example/golang.mk]: Sorry, we won't strip the symbol tables. They are very useful for debugging.

Comment (bmanojlovic) [reply example/golang.mk]: regarding strip sure, i am fine with it, maybe additional conditional is better way in any case

Comment (bmanojlovic) [reply example/golang.mk]: question regarding conditional is, do you use gnu make or BSD make (macos i have no clue what exactly have)

**Comment by bmanojlovic** [comment]
about what i used it for is for simple utility i already done in C, but i wanted to  make it a bit easier to maintain and clear bugs - so i re-wrote it in golang.
I have stumbled on your template by chance, as i was thinking on writing terraform provider  for graylog and found your project, and as it was empty but from some boilerplate code :)
here is how i used it
https://github.com/bmanojlovic/sendsms

File: example/golang.mk (hunk 1 above)
Comment (svenwltr) [reply example/golang.mk]: I think the easiest approach is to introduce a new variable (`ADDITIONAL_LDFLAGS`?), and define this in the `Makefile` and not in `golang.mk`. The Makefile should not fail if the variable isn't set.

**Comment by svenwltr** [comment]
Ok. Nice to hear and thanks for the effort. Please be aware, that this project might heavily change at any time, since it reflects the current best practices in our company.

**Comment by bmanojlovic** [comment]
i have just added conditional, can you see if that would be ok for you
```
steki@pc:~/GITHUB/sendsms> make
go build \
        -ldflags "  -X 'github.com/bmanojlovic/sendsms/cmd.BuildVersion=502a548.dirty' -X 'github.com/bmanojlovic/sendsms/cmd.BuildDate=Sat 10 Mar 13:35:32 CET 2018' -X 'github.com/bmanojlovic/sendsms/cmd.BuildHash=502a548215beb2466fd2a424d7c4e612e5fb230d' -X 'github.com/bmanojlovic/sendsms/cmd.BuildEnvironment=steki@pc.djule.org' " \
        -o sendsms-502a548.dirty-linux-amd64
ln -sf sendsms-502a548.dirty-linux-amd64 sendsms
steki@pc:~/GITHUB/sendsms> STRIP_BIN=yes make
go build \
        -ldflags " -s -w -X 'github.com/bmanojlovic/sendsms/cmd.BuildVersion=502a548.dirty' -X 'github.com/bmanojlovic/sendsms/cmd.BuildDate=Sat 10 Mar 13:35:39 CET 2018' -X 'github.com/bmanojlovic/sendsms/cmd.BuildHash=502a548215beb2466fd2a424d7c4e612e5fb230d' -X 'github.com/bmanojlovic/sendsms/cmd.BuildEnvironment=steki@pc.djule.org' " \
        -o sendsms-502a548.dirty-linux-amd64
ln -sf sendsms-502a548.dirty-linux-amd64 sendsms
steki@pc:~/GITHUB/sendsms> 
```

**Comment by svenwltr** [comment]
I still would prefer to add a new variable to the `ldflags`, so it can be specified in the actual `Makefile`.

```
BUILD_FLAGS=-ldflags "\
	$(ADDITIONAL_LDFLAGS) \
 	-X '$(PACKAGE)/cmd.BuildVersion=$(BUILD_VERSION)' \
	-X '$(PACKAGE)/cmd.BuildVersion=$(BUILD_VERSION)' \
 	-X '$(PACKAGE)/cmd.BuildDate=$(BUILD_DATE)' \	
```

You `Makefile` could look like this:

```
PACKAGE=github.com/bmanojlovic/sendsms
ADDITIONAL_LDFLAGS="-s -w"

include golang.mk
```

**Comment by bmanojlovic** [comment]
what do you think about this
`Makefile`

```
STRIP_BINARIES=yes
ADDITIONAL_LDFLAGS=-X cmd.version=1.0.0
```

`golang.mk`

```

ifneq ($(STRIP_BINARIES),)
ADDITIONAL_LDFLAGS+=-s -w
endif

BUILD_FLAGS=-ldflags "\
        $(ADDITIONAL_LDFLAGS) \
        -X '$(PACKAGE)/cmd.BuildVersion=$(BUILD_VERSION)' \
        -X '$(PACKAGE)/cmd.BuildDate=$(BUILD_DATE)' \
        -X '$(PACKAGE)/cmd.BuildHash=$(BUILD_HASH)' \
        -X '$(PACKAGE)/cmd.BuildEnvironment=$(BUILD_USER)@$(BUILD_MACHINE)' \
"
```


and of course example run

```
steki@pc:~/go/src/github.com/bmanojlovic/sendsms> make
steki@pc:~/go/src/github.com/bmanojlovic/sendsms> make
go build \
        -ldflags " -X cmd.version=1.0.0 -s -w -X 'github.com/bmanojlovic/sendsms/cmd.BuildVersion=502a548.dirty' -X 'github.com/bmanojlovic/sendsms/cmd.BuildEnvironment=steki@pc.djule.org' " \
        -o sendsms-502a548.dirty-linux-amd64
ln -sf sendsms-502a548.dirty-linux-amd64 sendsms

```

So why this way maybe, in any case it will add flags, but if you require stripping it will be done explicitly

**Comment by svenwltr** [comment]
I don't really understand why you want to put the `-s -w` flags into the `golang.mk`.

Also sorry for the conflicting files. We made a bigger change.

**Comment by bmanojlovic** [comment]
nice to have, but not forcing ;)
Anyway i have fixed conflicts

**Comment by svenwltr** [comment]
Thank you very much!

**Pull request merged**